embrace-android-delivery/src/main/kotlin/io/embrace/android/embracesdk/internal/delivery/scheduling/SchedulingServiceImpl.kt

```diff
@@ -66,13 +63,9 @@ class SchedulingServiceImpl(
     private fun startDeliveryLoop() {
         // When a payload arrives, check to see if there's already an active job try to deliver payloads
         // If not, schedule job. If so, do nothing.
-        if (sendLoopActive.compareAndSet(false, true)) {
-            deliveryTracer?.onStartDeliveryLoop(true)
-            schedulingWorker.submit {
-                deliveryLoop()
-            }
-        } else {
-            deliveryTracer?.onStartDeliveryLoop(false)
+        deliveryTracer?.onStartDeliveryLoop()
+        schedulingWorker.submit {
+            deliveryLoop()
         }
     }
 
```

Comment:
I attempted this but it became problematic as only taking the first payload means it would be necessary to track state for payloads that were attempted & weren't attempted. Without adding that state I simply ran into GC problems as the proposed change would loop indefinitely

Reply:
So the in-progressed payload should be filtered out in with the filter on `activeSends` inside `StoredTelemetryMetadata.shouldSendPayload()`, which is called inside `createPayloadQueue`. Doesn't that happen if you simply changed `createPayloadQueue` to return the whole LinkedList but only the first element?

Reply:
e.g. change that function to:

```
private fun createPayloadQueue(exclude: Set<StoredTelemetryMetadata> = emptySet()): StoredTelemetryMetadata? {
        val payloadsByPriority = storageService.getPayloadsByPriority()
        val payloadToSend = payloadsByPriority
            .filter { it.shouldSendPayload() && !exclude.contains(it) }
            .sortedWith(storedTelemetryComparator)
            .first()
        deliveryTracer?.onPayloadQueueCreated(
            payloadsByPriority,
            payloadToSend,
        )
        return payloadToSend
    }

Reply:
It doesn't work for the case where a payload isn't enqueued (e.g. if an endpoint is rate-limited) and hangs the application. We could track those payloads too but I'm wary that might be adding more complexity on top. We can chat synchronously about options later today

Reply:
Disregard my comment above - I think I just made a mistake in the original implementation of these changes & things now work fine on my latest iteration. There are only 2 states I can see (failed vs active) so this approach should work fine.